src/core/utils/helpers/general.ts

```diff
@@ -345,4 +345,28 @@ export const dataIsNotEmpty = (data: unknown[]) => Boolean(data.length);
 export const constructModalId = (prefix: string, fragments: string[]) =>
   `${prefix ? `${prefix}-` : ""}${fragments.join("-")}`;
 
+export const getReleaseYear = (work: Work) => {
+  const existsOnWork = work.fictionNonfiction;
+  const { latest, bestRepresentation } = work.manifestations;
+  switch (
+    materialIsFiction(existsOnWork ? work : bestRepresentation || latest)
+  ) {
+    case true:
+      return (
+        work.workYear?.year ||
+        bestRepresentation.dateFirstEdition?.display ||
+        bestRepresentation.dateFirstEdition?.year ||
+        null
+      );
+    case false:
+      return (
+        latest.edition?.publicationYear?.display ||
+        latest.workYear?.year ||
+        null
+      );
+    default:
+      return null;
+  }
+};
```

Comment:
I don't think it makes sense to use a switch case here.
`materialIsFiction` is always returning a `boolean` and the default case would never be executed.
Suggestion (har ikke testet:
```typescript
export const getReleaseYear = (work: Work) => {
  const existsOnWork = work.fictionNonfiction;
  const { latest, bestRepresentation } = work.manifestations;
  const manifestation = bestRepresentation || latest;

  if (existsOnWork && materialIsFiction(work)) {
    return work.workYear?.year || null;
  }

  if (materialIsFiction(manifestation)) {
    return (
      manifestation.dateFirstEdition?.display ||
      manifestation.dateFirstEdition?.year ||
      null
    );
  }

  return (
    latest.edition?.publicationYear?.display || latest.workYear?.year || null
  );
};

Reply:
Hov har faktisk en bedre version, hvor vi sikrer os at return er: `string | null`:
```typescript
export const getReleaseYear = (work: Work) => {
  const { latest, bestRepresentation } = work.manifestations;
  const manifestation = bestRepresentation || latest;

  // If the work tells us that it is fiction.
  if (materialIsFiction(work)) {
    return stringifyValue(work.workYear?.year) || null;
  }

  // If the manifestation tells us that it is fiction.
  if (materialIsFiction(manifestation)) {
    return (
      stringifyValue(
        manifestation.dateFirstEdition?.display ||
        manifestation.dateFirstEdition?.year
      ) || null
    );
  }

  // Otherwise try to
  // use the publication year of the latest manifestation edition.
  // or use the work year of the latest manifestation.
  // or surrender with a null.
  return (
    stringifyValue(
      latest.edition?.publicationYear?.display || latest.workYear?.year
    ) || null
  );
};
```
Men det kræver en ændring af `stringifyValue()`:
```typescript
export const stringifyValue = (value: number | string | null | undefined) =>
  value ? String(value) : "";

Reply:
I like your other solution :) 
But without stringifying the value. It is possibly null/undefined, we don't want that rendered as a string.